ui/articles-aem-renderer/src/main/java/org/cru/godtools/articles/aem/fragment/AEMArticleItemFragment.java

```diff
@@ -142,13 +152,63 @@ private String injectCss(@NonNull final String content) {
         @Nullable
         @Override
         public WebResourceResponse shouldInterceptRequest(WebView view, String url) {
-            if (url.endsWith(".jpg")) {
-                //TODO: Load local Image WebResource
-            } else if (url.endsWith(".css")) {
-                //TODO: Load CSS WebResource
+            String extension = MimeTypeMap.getFileExtensionFromUrl(url);
+
+            WebResourceResponse response = null;
+
+            String mimeType;
+
+            switch (extension) {
+                case "jpg":
+                case "png":
+                case "bmp":
+                case "gif":
+                    mimeType = String.format("image/%s", "jpg".equals(extension) ? "jpeg" : extension);
+                    response = getResponseFromFile(mimeType, url);
+                    break;
+                case "css":
+                    mimeType = "text/css";
+                    response = getResponseFromFile(mimeType, url);
+                    break;
+                default:
+                    mimeType = "application/octet-stream";
+                    response = getResponseFromFile(mimeType, url);
+            }
```

Comment:
your mime-type resolution logic can be simplified. create a constant map that lists extensions with their mimetype.
```java
private static final Map<String, String> MIME_TYPES = ImmutableMap.of(
    "gif", "image/gif",
    "css", "text/css",
    ...
);
```

Then just do a lookup by the extension:
```java
String mimeType = MIME_TYPES.get(extension);
if(mimeType == null) {
    mimeType = "application/octet-stream";
}
```